Compilation error for `mklgpu/mklgpu_batch.cpp`

# Summary

Using the name “MAJOR” here https://github.com/oneapi-src/oneMKL/blob/d06919ca0f1c675b170df94259f319eb1d020d5c/src/blas/backends/mklgpu/mklgpu_batch.cxx#L299 causes compiling problems with the version of DPC++ I have (`Intel(R) oneAPI DPC++/C++ Compiler 2022.1.0 (2022.x.0.20211025)`). It looks like the issue is because there’s a class named MAJOR which becomes MKL_COL_MAJOR from the `#define MAJOR MKL_COL_MAJOR` and a enum member defined as MKL_COL_MAJOR https://github.com/oneapi-src/oneMKL/blob/d06919ca0f1c675b170df94259f319eb1d020d5c/src/blas/backends/mklgpu/mklgpu_common.hpp#L26

As work around I changed `class MAJOR` to “class test” and it compiled fine. If there's a better workaround or I messed something else up, let me know!

# Version
oneMKL git commit `d06919ca0f1c675b170df94259f319eb1d020d5c`

# Environment
* HW you use: Intel GPU
* Backend library version:  MKL
* OS name and version: openSUSE 15.3
* Compiler version: DPC++ `Intel(R) oneAPI DPC++/C++ Compiler 2022.1.0 (2022.x.0.20211025)`
* CMake output log: See output below

# Steps to reproduce
Please check that the issue is reproducible with the latest revision on
master. Include all the steps to reproduce the issue.

```
    git clone https://github.com/oneapi-src/oneMKL.git
    cd oneMKL
    rm -rf build
    mkdir build
    cd build
    CXX=`which dpcpp` cmake ../ -DMKL_ROOT=$MKLROOT -DREF_BLAS_ROOT=${blas_root} -DREF_LAPACK_ROOT=${blas_root} -DSYCL_LIBRARY=${SDK_ROOT}/compiler/latest/linux/lib/libsycl.so
    cmake --build . -j1
```

# Observed behavior
When I try to build it fails with:

```
[  4%] Building CXX object bin/blas/backends/mklgpu/CMakeFiles/onemkl_blas_mklgpu_obj.dir/mklgpu_batch.cpp.o
In file included from /gpfs/jlse-fs0/users/bertoni/oneMKL/oneMKL/src/blas/backends/mklgpu/mklgpu_batch.cpp:34:
/gpfs/jlse-fs0/users/bertoni/oneMKL/oneMKL/src/blas/backends/mklgpu/mklgpu_batch.cxx:84:50: error: 'MKL_COL_MAJOR' does not refer to a value
    ::oneapi::mkl::gpu::sgemv_batch_sycl(&queue, MAJOR, mkl_convert(transa), m, n, alpha, &a, lda,
                                                 ^
/gpfs/jlse-fs0/users/bertoni/oneMKL/oneMKL/src/blas/backends/mklgpu/mklgpu_batch.cpp:33:15: note: expanded from macro 'MAJOR'
#define MAJOR MKL_COL_MAJOR
              ^
/tmp/mklgpu_batch-header-7d3491.h:8:7: note: declared here
class MKL_COL_MAJOR;
      ^

```

# Expected behavior
I expect it to compile.

--- a/src/blas/backends/cublas/cublas_scope_handle.cpp
+++ b/src/blas/backends/cublas/cublas_scope_handle.cpp
@@ -31,7 +31,8 @@ namespace cublas {
  * takes place if no other element in the container has a key equivalent to
  * the one being emplaced (keys in a map container are unique).
  */
-thread_local cublas_handle<pi_context> CublasScopedContextHandler::handle_helper = cublas_handle<pi_context>{};
+thread_local cublas_handle<pi_context> CublasScopedContextHandler::handle_helper =
+    cublas_handle<pi_context>{};
 
 CublasScopedContextHandler::CublasScopedContextHandler(cl::sycl::queue queue,
                                                        cl::sycl::interop_handler &ih)

--- a/src/blas/backends/mklgpu/mklgpu_batch.cpp
+++ b/src/blas/backends/mklgpu/mklgpu_batch.cpp
@@ -30,16 +30,20 @@ namespace blas {
 namespace mklgpu {
 namespace column_major {
 
-#define MAJOR MKL_COL_MAJOR
+#define MAJOR             MKL_COL_MAJOR
+#define EMPTY_KERNEL_NAME ColMajorEmptyKernel
 #include "mklgpu_batch.cxx"
 #undef MAJOR
+#undef EMPTY_KERNEL_NAME
 
 } // namespace column_major
 namespace row_major {
 
-#define MAJOR MKL_ROW_MAJOR
+#define MAJOR             MKL_ROW_MAJOR
+#define EMPTY_KERNEL_NAME RowMajorEmptyKernel
 #include "mklgpu_batch.cxx"
 #undef MAJOR
+#undef EMPTY_KERNEL_NAME
 
 } // namespace row_major
 } // namespace mklgpu

--- a/src/blas/backends/mklgpu/mklgpu_batch.cxx
+++ b/src/blas/backends/mklgpu/mklgpu_batch.cxx
@@ -296,7 +296,7 @@ cl::sycl::event *coalesce_events(cl::sycl::queue &queue, std::vector<cl::sycl::e
         return new cl::sycl::event(queue.submit([&](cl::sycl::handler &cgh) {
             for (int64_t i = 0; i < prereqs.size(); i++)
                 cgh.depends_on(*prereqs[i]);
-            cgh.single_task<class MAJOR>([]() {});
+            cgh.single_task<class EMPTY_KERNEL_NAME>([]() {});
         }));
     }
     else

